content/tutorials/history.textile

```diff
@@ -31,17 +31,67 @@ We will be using channel rules in this tutorial to ensure all channels in the @p
 
 You have now enabled history for all channels in the @persisted@ namespace i.e. any channel with a name matching the pattern @peristed:*@ will store published messages to disk.
 
-h2. Step 3 - Install Ably
+h2. 
+  default: Step 3 – Install Ably
+  android: Step 3 – Set up environment and install Ably
 
 blang[java].
+  To start using Ably in your Java application, you need to include the Ably Client library. We recommend that you include the latest client library via Gradle in your project's @gradle.build@ file.
+  
+  ```[groovy]
+  apply plugin: 'application'
+  mainClassName = 'io.ably.tutorial.Example'
+
+  repositories {
+    jcenter()
+      maven { url "https://raw.github.com/paddybyers/Java-WebSocket/mvn-repo/" }
+  }
+
+  dependencies {
+      compile 'io.ably:ably-java:0.8.4'
+  }
+  ```
+
+  After you add neccessary dependencies, you can import AblyRest class into your code and initialize it.
+
+  ```[java]
+  import io.ably.lib.rest.AblyRest;
+
+  public class Example {
+    private final static String API_KEY = "INSERT-YOUR-API-KEY-HERE"; 
+    /* RuntimeException will be thrown if API_KEY will not be set to a proper one */
+    static {
+        if (API_KEY.contains("INSERT")) {
+            throw new RuntimeException("API key is not set, sign up at ably.io to get yours");
+        }
+    }
+
+    public static void main(String[] args) {
+        try {
+          initAbly();
+        } catch (AblyException e) {
+          e.printStackTrace();
+        }
+    }
+
+    private static void initAbly() throws AblyException {
+        AblyRest ablyRest = new AblyRest(API_KEY);
+    }
+  }
+  ```
+
+  "See this step in Github":https://github.com/piotrekinventica/tutorials/commit/9dbbcf3
```

Comment:
Why are you referencing your repo?  This is all published at ably/tutorials?  You have push access I believe to that repo anyway, but the SHA's wont change so this is not necessary regardless.